hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -629,16 +649,23 @@ private Set<String> getOrCreatePendingClusteringInstantSet() {
   /**
    * Returns the first non savepoint commit on the timeline.
    */
-  private static Option<HoodieInstant> findFirstNonSavepointCommit(List<HoodieInstant> instants) {
+  private static Option<HoodieInstant> findFirstNonSavepointCommit(
+      List<HoodieInstant> instants,
+      boolean byCompletionTime) {
     Set<String> savepointTimestamps = instants.stream()
         .filter(entry -> entry.getAction().equals(HoodieTimeline.SAVEPOINT_ACTION))
-        .map(HoodieInstant::getTimestamp)
+        .map(byCompletionTime
+            ? HoodieInstant::getCompletionTime
+            : HoodieInstant::getTimestamp)
         .collect(Collectors.toSet());
     if (!savepointTimestamps.isEmpty()) {
       // There are chances that there could be holes in the timeline due to archival and savepoint interplay.
       // So, the first non-savepoint commit is considered as beginning of the active timeline.
       return Option.fromJavaOptional(instants.stream()
-          .filter(entry -> !savepointTimestamps.contains(entry.getTimestamp()))
+          .filter(entry -> !savepointTimestamps.contains(
```

Comment:
This still assumes that the `instants` is sorted by completion time to give the correct results, but the passed-in `instants` is always sorted by instant time.

Reply:
Fixed.